src/Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch/ElasticSearchOutput.cs

```diff
@@ -378,10 +379,25 @@ private async Task EnsureIndexExists(string indexName, ElasticClient esClient)
             }
         }
 
-        private string GetIndexName(ElasticSearchConnectionData connectionData)
+        protected internal string GetIndexName(ElasticSearchConnectionData connectionData)
         {
-            DateTimeOffset now = DateTimeOffset.UtcNow;
-            string retval = connectionData.Configuration.IndexNamePrefix + now.ToString("yyyy" + Dot + "MM" + Dot + "dd");
+            string dateString = "";
+            try
+            {
+                dateString = string.Format($"{{0:{connectionData.Configuration.IndexFormat}}}", DateTimeOffset.UtcNow);
+            }
+            catch (Exception ex)
+            {
+                string errorMessage = $"{nameof(ElasticSearchOutput)}: indexFormat could not parse as date: {ex.Message}{Environment.NewLine}" +
+                                      $"Invalid IndexFormat value: {connectionData.Configuration.IndexFormat}{Environment.NewLine}" +
+                                      $"Falling back to default value: {ElasticSearchOutputConfiguration.DefaultIndexFormat}";
+                // Changing in connectionData.Configuration so we don't report this every log message
+                connectionData.Configuration.IndexFormat = ElasticSearchOutputConfiguration.DefaultIndexFormat;
+                healthReporter.ReportProblem(errorMessage);
```

Comment:
I would use a warning here

Reply:
I have also added new format strings to enable Quarter of Year, Week of Month, and WeekOfYear. Can I add that update at the same time, or should it be separate? It is quite a bit more complex than the previous update, so I can see there being more objections to it that this first PR.

Reply:
Typically, with a PR, you would just add more commits to the repository/branch that you used to create the PR. These commits will then be "included" in the PR.  For this PR it would be `BuddhaBuddy1/diagnostics-eventflow:master` repository/branch.

Reply:
Also, the new format strings sound very much related to the rest of the enhancement that you are making, so please, go ahead and add them and I will re-review, no problem, and thank you!

Reply:
Changed to warning, new formatting added. If anyone knows how to make the readme changes prettier, feel free. :-)

Reply:
I feel like I made a mistake by merging my changes into my fork's master first, then doing a PR on that. Should I have left it in the branch?

Reply:
It is not a huge deal, but the typical workflow that I use is that you create a separate branch for the PR. Then you can use that branch to add more commits to it and/or decide what other changes to merge into that branch. It isolates the work you are doing on the PR from your other work, and from changes that come from upstream repository too.